### ISSUE TITLE
TP 0 - Parte 4: Problema con sockets

### ISSUE THREAD

**Issue by ValenSampayo** [sisoputnfrba/foro#3400]
Buenas, ya leí la guía de sockets pero no encontre mención sobre este problema y tampoco encontré un issue que haya pasado lo mismo. El problema es que se queda "Colgado" ambas partes, el cliente y el servidor. El server se queda colgado esperando el cliente y el cliente se queda colgado esperando el accept por parte del server. Estuve revisando con el debug los valores y el socket del cliente y el servidor tienen el mismo valor (4). 

Al final de esto me termina tirando un error "Broken pipe" al tratar de hacer el send desde el cliente ¿Puede que sea porque esta tratando de enviar el mensaje al mismo socket o es otra la causa de este error?.

Valores vistos con el debug del server:
![image](https://github.com/sisoputnfrba/foro/assets/49347551/20ae91f3-be78-440f-b8b3-f1b603bf541c)

Valores vistos con el debug del cliente:
![image](https://github.com/sisoputnfrba/foro/assets/49347551/e73572ec-7aa4-45c8-963e-55ea4362b203)

Broken Pipe exception:
![image](https://github.com/sisoputnfrba/foro/assets/49347551/c4dd326c-b219-44de-97e6-811c5f752784)

**Comment by iago64** [comment]
Buenas! Cómo va?

Solo para entender un poco el contexto, cuando inicias el servidor llegas al `accept()` y se queda esperando ahi a que se conecte algo? 
La función de `iniciar_servidor()` como te quedó implementada?

Saludos.-

**Comment by ValenSampayo** [comment]
Claro, el servidor llega hasta la función de esperar_cliente que es donde está el accept y ahí queda colgado incluso después de que yo inicio el cliente y trato de mandar el mensaje.

La función iniciar_servidor() está asi:
int iniciar_servidor(void)
{
	int socket_servidor;

	struct addrinfo hints, *servinfo, *p;

	memset(&hints, 0, sizeof(hints));
	hints.ai_family = AF_UNSPEC;
	hints.ai_socktype = SOCK_STREAM;
	hints.ai_flags = AI_PASSIVE;

	//getaddrinfo(NULL, PUERTO, &hints, &servinfo);
	int result = getaddrinfo(NULL, PUERTO, &hints, &servinfo); //Devuelve 0 cuando esta todo OK
		if (result != 0){
			log_error(logger,"Error en getaddrinfo_: %s", gai_strerror(result));
			exit(-1);
		}

	// Creamos el socket de escucha del servidor
		socket_servidor = socket(servinfo->ai_family,
								servinfo->ai_socktype,
								servinfo->ai_protocol); //Si devuelve -1 hubo error creandolo
		if(socket_servidor == -1){
			log_error(logger, "Error en socket: %s", strerror(socket_servidor));
   			exit(-2);
		} 

	// Asociamos el socket a un puerto
		int bindRes = bind(socket_servidor,servinfo->ai_addr,servinfo->ai_addrlen);
		
		if(bindRes != 0){
			error("Fallo el bind");
			exit(-3);
		}
/*
		if(bind(socket_servidor,(void*) &servinfo,sizeof(servinfo))!= 0){
			perror("Fallo el bind");
			exit(-3);
		} else{
			log_info(logger,"Se hizo el bind correctamente");
		}
*/
		
	// Escuchamos las conexiones entrantes
		listen(socket_servidor,SOMAXCONN);
		
	freeaddrinfo(servinfo);
	log_trace(logger, "Listo para escuchar a mi cliente");

	return socket_servidor;
}

La función esperar_cliente() está así: 
int esperar_cliente(int socket_servidor)
{
	log_info(logger,"Esperando cliente...");
	// Aceptamos un nuevo cliente
	int socket_cliente;

		socket_cliente = accept(socket_servidor, NULL, NULL);

	log_info(logger, "Se conecto un cliente!");

	return socket_cliente;
}

**Comment by iago64** [comment]
Buenas!...

Acabo de darme cuenta del error despues de ponerme a repasar el código del TP0 paso a paso porque no me cerraba el problema y veo que la IP que tenes logueada del lado cliente es **172**.0.0.1 cuando debería ser 127.0.0.1 que es la IP de localhost.

Definitivamente, como dice la frase: El diablo esta en los detalles...

Saludos.-

**Comment by ValenSampayo** [comment]
Ahí quedo funcionando todo gracias, cierro el issue.

Saludos
